### ISSUE TITLE
Page rendering speed needs improving

### ISSUE THREAD

**Issue by dancgray** [adaptlearning/adapt_authoring#2450]
## Affected Area
Page editing

## Requested Feature
Loading times for the page editor has some noticeable latency on rendering the components. This is amplified by the number of components you have on the page. 

## Use Case
Experience editing a page is negatively affected waiting >7 seconds for a page with 18 components to render. 

## Current Workaround
None

## Additional Information
There are multiple calls to /api/content/component?_parentId with at least two models created for each component.

The additional calls to /api/content/component?_parentId are created when the layout options are [evaluated](https://github.com/adaptlearning/adapt_authoring/blob/master/frontend/src/modules/editor/contentObject/views/editorPageComponentView.js#L160) for each component.

Additionally it may be possible to reduce server side latency when an [additional database query ](https://github.com/adaptlearning/adapt_authoring/blob/master/lib/contentmanager.js#L815) is made just to retrieve the component type name.

AT v0.10.0
Node v10.15.3

**Comment by taylortom** [comment]
The only sensible way I can see to improve the number of requests is to introduce extra fields to the component mongo doc (e.g. page ID). Either that or introduce a new endpoint to collate the different results.

**Comment by dancgray** [comment]
Agreed. We could however pass the sibling count down to the component [here](https://github.com/adaptlearning/adapt_authoring/blob/master/frontend/src/modules/editor/contentObject/views/editorPageBlockView.js#L217) by adding the count to the component model `this.children[i].siblingCount = this.children.length`.

Would halve the calls to api/content/component?_parentId and number of component models but might not reduce load times that much.
